### PR TITLE
CS169 Iteration 1 Pull Request

### PR DESCRIPTION
**Pull request by ArushC**
### [Pivotal Tracker Link][tracker]

[tracker]: https://www.pivotaltracker.com/story/show/your-story-id

We decided to include the links to each of the pull requests that were made over the course of this iteration. See below. PRs are listed in chronological order, from earliest to most recent:

Change reason to denial reason: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/29](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/29)

Add to field: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/30](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/30)

Add email resend flash message: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/31](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/31)

Add request more info feature as admin: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/32](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/32)

Edit redirect to use current path: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/33](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/33)

Fix email template reseed problem: [https://github.com/cs169/BJC-Teacher-Tracker-App/pull/34](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/34)

## What this PR does:

Implements iteration 1 features, including request more information button, the "to" field in the email template, and the flash message on resending emails. We also changed "reason" to "denial_reason" and fixed the duplicate email entry problem that occurs when you run "rails db:seed" after running migrations.

#### Who authored this PR?

Everyone.

### How should this PR be tested?

See cucumber and RSpec tests, and view the latest deployment at [https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/dashboard](https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/dashboard).

* Is there a deploy we can view?
[https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/dashboard](https://sp24-01-bjc-teachers-d0e972207d82.herokuapp.com/dashboard)
* What do the specs/features test?
To make sure everything works as expected
* Are there edge cases to watch out for?
No

#### Are there any complications to deploying this?

No

Run "rails db:migrate" and "rails db:seed".

### Checklist:

- [ ] Has this been deployed to a staging environment or reviewed by a customer?
- [ ] Tag someone for code review (either a coach / team member)
- [ ] I have renamed the branch to match PivotTracker's suggested one (necessary for BlueJay) (e.g. `michael/12345-add-new-feature` Any branch name will do as long as the story ID is there. You can use `git checkout -b [new-branch-name]`)

### CODE REVIEW

**Comment by perryzjc** [comment]
A [new PR](https://github.com/cs169/BJC-Teacher-Tracker-App/pull/35) just merged into cs169's main branch, which fixes the implementation & tests for email reseed problem we discussed last Friday (Mar 1).

**Comment by fp456** [comment]
Looks good so far but we can add more testing, especially for "Add request more info feature as admin".

**Review by cycomachead — CHANGES_REQUESTED** [review]
Some initial comments

File: db/migrate/20240218005604_add_to_field_to_email_template.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddToFieldToEmailTemplate < ActiveRecord::Migration[6.1]
+  def change
+    add_column :email_templates, :to, :string
+    #below is the default prepopulated 'to' field value,
```
[the remark this answers is not in the document]
Comment (perryzjc) [reply db/migrate/20240218005604_add_to_field_to_email_template.rb:4]: During the class discussion, Michael thought we could just skip the migration for this part; he could just manually update the field.
